### PR TITLE
psp-9148 Add consultation outcomes and business logic.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: source/backend/entities/PimsBaseContext.cs [hunk 1]
```diff
@@ -1698,6 +1704,38 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
 
+        modelBuilder.Entity<PimsConsultationOutcomeType>(entity =>
+        {
+            entity.HasKey(e => e.ConsultationOutcomeTypeCode).HasName("OUTCMT_PK");
+
+            entity.ToTable("PIMS_CONSULTATION_OUTCOME_TYPE", tb =>
+                {
+                    tb.HasComment("Description of the consultation outcome type for a lease or license.");
+                    tb.HasTrigger("PIMS_OUTCMT_I_S_I_TR");
+                    tb.HasTrigger("PIMS_OUTCMT_I_S_U_TR");
```
Comment (FuriousLlama) [inline source/backend/entities/PimsBaseContext.cs:1715]: why is the context being updated?

File: source/backend/entities/ef/PimsConsultationOutcomeType.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+using System.ComponentModel.DataAnnotations.Schema;
+using Microsoft.EntityFrameworkCore;
+
+namespace Pims.Dal.Entities;
+
+/// <summary>
+/// Description of the consultation outcome type for a lease or license.
+/// </summary>
+[Table("PIMS_CONSULTATION_OUTCOME_TYPE")]
+public partial class PimsConsultationOutcomeType
```
Comment (FuriousLlama) [inline source/backend/entities/ef/PimsConsultationOutcomeType.cs:13]: Should this have been part of the scaffold?

File: source/backend/tests/unit/api/Controllers/Leases/ConsultationControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            // Arrange
+            var consultation = new PimsLeaseConsultation() { LeaseId = 1 };
+
+            this._service.Setup(m => m.GetConsultationById(It.IsAny<long>())).Returns(consultation);
+
+            // Act
+            var result = this._controller.GetLeaseConsultationById(1, 1);
+
+            // Assert
+            this._service.Verify(m => m.GetConsultationById(It.IsAny<long>()), Times.Once());
```
Comment (FuriousLlama) [inline source/backend/tests/unit/api/Controllers/Leases/ConsultationControllerTest.cs:55]: not a big fan of these tests. It leaks implementation details about what the controller does internally and does not provide a validation of anything other than proxying the call

File: source/backend/entities/PimsBaseContext.cs (hunk 1 above)
Comment (devinleighsmith) [reply source/backend/entities/PimsBaseContext.cs:1715]: Oh, this was branched from before the scaffold was merged, I'll rebase.

File: source/backend/tests/unit/api/Controllers/Leases/ConsultationControllerTest.cs (hunk 3 above)
Comment (devinleighsmith) [reply source/backend/tests/unit/api/Controllers/Leases/ConsultationControllerTest.cs:55]: This is our pattern thus far for these tests, which we have gone through together as a group. Recommend we bring this up for review in our dev meeting, as nearly all of our controllers are tested this way.

File: source/frontend/src/features/mapSideBar/lease/tabs/consultations/edit/EditConsultationYupSchema.ts [hunk 4]
```diff
@@ -1,14 +1,26 @@
 /* eslint-disable no-template-curly-in-string */
 import * as Yup from 'yup';
 
+import { ApiGen_CodeTypes_ConsultationOutcomeTypes } from './../../../../../../models/api/generated/ApiGen_CodeTypes_ConsultationOutcomeTypes';
```
[the remark this answers is not in the document]
Comment (devinleighsmith) [reply source/frontend/src/features/mapSideBar/lease/tabs/consultations/edit/EditConsultationYupSchema.ts:4]: hmm, I use a plugin to generate these, I wonder if I can configure it to use absolute paths by default.

**Review by FuriousLlama — APPROVED** [review]

**Review by asanchezr — APPROVED** [review]

**Pull request merged**